src/js/pages/TwitterHandleLanding.jsx

```diff
@@ -52,36 +52,19 @@ export default class TwitterHandleLanding extends Component {
   }
 
   // eslint-disable-next-line camelcase,react/sort-comp
-  UNSAFE_componentWillReceiveProps (nextProps) {
-    // console.log('TwitterHandleLanding componentWillReceiveProps');
-    let activeRoute = '';
-    let nextTwitterHandle;
-    // eslint-disable-next-line react/prop-types
-    if (nextProps.match && nextProps.location) {
-      // eslint-disable-next-line react/prop-types,camelcase
-      const { location: { pathname }, match: {params: { twitter_handle} }  } = nextProps;
-      activeRoute = pathname;
-      nextTwitterHandle = twitter_handle;
-      // eslint-disable-next-line react/prop-types
-    } else if (this.props && this.props.location) {
-      // eslint-disable-next-line react/prop-types
-      const { location: { pathname } } = this.props;
-      activeRoute = pathname;
+  componentDidUpdate(prevProps) {
+    const { match: { params: { twitter_handle: twitterHandle } }, location: { pathname: activeRoute } } = this.props;
+    // if-statements prevents infinite rendering
+    if (prevProps.location.pathname !== this.props.location.pathname) {
+      this.setState({activeRoute});
     }
-
-    const { twitterHandle } = this.state;
-    this.setState({
-      activeRoute,
-    });
-    // console.log('TwitterHandleLanding componentWillReceiveProps activeRoute:', activeRoute);
-    if (nextTwitterHandle && twitterHandle.toLowerCase() !== nextTwitterHandle.toLowerCase()) {
-      // We need this test to prevent an infinite loop
-      console.log('TwitterHandleLanding componentWillReceiveProps, different twitterHandle: ', nextProps.match.params.twitter_handle);
-      TwitterActions.resetTwitterHandleLanding();
-      TwitterActions.twitterIdentityRetrieve(nextTwitterHandle);
-      this.setState({
-        twitterHandle: nextTwitterHandle,
-      });
+    if (prevProps.match.params.twitter_handle !== this.props.match.params.twitter_handle) {
```

Comment:
Please make sure both versions of the twitter_handle are compared in lower case.